akka-stream/src/main/scala/akka/stream/javadsl/Flow.scala

```diff
@@ -1315,6 +1315,32 @@ final class Flow[In, Out, Mat](delegate: scaladsl.Flow[In, Out, Mat]) extends Gr
       d: java.time.Duration): javadsl.Flow[In, java.util.List[Out], Mat] =
     groupedWeightedWithin(maxWeight, costFn, d.asScala)
 
+  /**
+   * Chunk up this stream into groups of elements received within a time window,
+   * or limited by the weight and number of the elements, whatever happens first.
+   * Empty groups will not be emitted if no elements are received from upstream.
+   * The last group before end-of-stream will contain the buffered elements
+   * since the previously emitted group.
+   *
+   * '''Emits when''' the configured time elapses since the last group has been emitted or weight limit reached
+   *
+   * '''Backpressures when''' downstream backpressures, and buffered group (+ pending element) weighs more than
+   * `maxWeight` or has more than `maxNumber` elements
+   *
+   * '''Completes when''' upstream completes (emits last group)
+   *
+   * '''Cancels when''' downstream completes
+   *
+   * `maxWeight` must be positive, `maxNumber` must be positive, and `d` must be greater than 0 seconds,
+   * otherwise IllegalArgumentException is thrown.
+   */
+  def groupedWeightedWithin(
+      maxWeight: Long,
+      maxNumber: Int,
+      costFn: function.Function[Out, java.lang.Long],
+      d: java.time.Duration): javadsl.Flow[In, java.util.List[Out], Mat] =
+    new Flow(delegate.groupedWeightedWithin(maxWeight, maxNumber, d.asScala)(costFn.apply).map(_.asJava))
```

Comment:
I know we are using `d` in other methods, but why are we being try to be so concise? 

I think that from the user, at least Scala users, perspective it will be better if we call it `duration`. Because then one could write code like: 


```scala
Source
 .fromPublisher(upstream)
  .groupedWeightedWithin(maxWeight = 10, maxNumber = 3, duration = 30.seconds) { elt => 
	// calculate cost
  }
```

If we use `d` that will look like:

```scala
  .groupedWeightedWithin(maxWeight = 10, maxNumber = 3, d = 30.seconds)
```

and that's weird.

I don't know what is the policy for param names. We probably can't rename in the other methods, but maybe better not to introduce new methods as such.

Reply:
@johanandren / @raboof 
Have you had a chance to take a look at this point? It is a one-minute change, however, we need to decide whether it is worth to do.

Reply:
This is java DSL and in Java you don't have named params anyway, so I don't think that is an argument, in general  for readability I think the public API should have clear names for sure, but given that there are more precedents here, let's leave it be and fix that separately.

Reply:
Created https://github.com/akka/akka/issues/30115 to track the parameter names